modular_zubbers/code/modules/bitrunning/disks.dm

```diff
@@ -0,0 +1,50 @@
+/obj/item/bitrunning_disk/prefs
+	name = "DeForest biological simulation disk"
+	desc = "A disk containing the biological simulation data necessary to load custom characters into bitrunning domains."
+	icon = 'icons/obj/devices/circuitry_n_data.dmi'
+	base_icon_state = "datadisk"
+	icon_state = "datadisk0"
+
+	w_class = WEIGHT_CLASS_SMALL
+
+	var/datum/preferences/loaded_preference
+
+	var/include_loadout = FALSE
+
+/obj/item/bitrunning_disk/prefs/examine(mob/user)
+	. = ..()
+	if(!isnull(loaded_preference))
+		var/name = loaded_preference.read_preference(/datum/preference/name/real_name)
+		. += "It currently has the character [name] loaded, with loadouts [(include_loadout ? "enabled" : "disabled")]"
+		. += span_notice("Ctrl-Click to change loadout loading")
+
+/obj/item/bitrunning_disk/prefs/CtrlClick(mob/user)
+	. = ..()
+	include_loadout = !include_loadout // We just switch this around. Elegant!
+	balloon_alert(user, include_loadout ? "Loadout enabled" : "Loadout disabled")
+
+/obj/item/bitrunning_disk/prefs/attack_self(mob/user, modifiers)
+	. = ..()
+
+	var/list/prefdata_names = user.client.prefs?.create_character_profiles()
```

Comment:
```suggestion
	var/list/prefdata_names = user.client.prefs?.create_character_profiles()
	if(!prefdata_names)
		return
```
I don#t think we can do anything useful if prefs is null?

Reply:
Technically there isn't a case where this can return null anyhow, but sure

Reply:
(There is the case of prefs being null- you used `prefs?.`..., which I believe returns null if prefs is null